actionpack/lib/action_dispatch/testing/assertions/response.rb

```diff
@@ -50,12 +50,19 @@ def assert_response(type, message = nil)
       #
       #   # Asserts that the redirection matches the regular expression
       #   assert_redirected_to %r(\Ahttp://example.org)
-      def assert_redirected_to(options = {}, message = nil)
-        assert_response(:redirect, message)
-        return true if options === @response.location
+      #
+      #   # Asserts that the redirection has the HTTP status code 301 (Moved
+      #   # Permanently).
+      #   assert_redirected_to "/some/path", status: :moved_permanently
+      def assert_redirected_to(url_options = {}, options = {}, message = nil)
+        options, message = message, nil if message.is_a?(Hash) && options.empty?
+
+        status = options[:status] || :redirect
```

Comment:
```suggestion
      def assert_redirected_to(url_options = {}, message = nil, status: nil)
```
Here's what I was suggesting.

Reply:
Thanks, I understand, but I was implementing the original feedback: https://github.com/rails/rails/pull/46057#discussion_r973678612

Are you now saying you prefer the keyword argument? Either approach works for me.

FWIW, the keyword approach has a backward compatibility issue with code like the following. Is that okay?

```
actionpack/test/controller/redirect_test.rb
277      assert_redirected_to action: "other_host", only_path: false, host: "other.test.host"
```

```
ArgumentError: unknown keywords: :action, :only_path, :host
```

Reply:
Ah my bad.